api/v1/nificluster_types.go

```diff
@@ -100,11 +100,18 @@ type NifiClusterSpec struct {
 	NifiControllerTemplate *string `json:"nifiControllerTemplate,omitempty"`
 	// ControllerUserIdentity specifies what to call the static admin user's identity
 	// Warning: once defined don't change this value either the operator will no longer be able to manage the cluster
-	ControllerUserIdentity *string `json:"controllerUserIdentity,omitempty"`
+	ControllerUserIdentity  *string                 `json:"controllerUserIdentity,omitempty"`
+	SingleUserConfiguration SingleUserConfiguration `json:"singleUserConfiguration,omitempty"`
 
 	// @TODO: Block Controller change
 }
 
+type SingleUserConfiguration struct {
+	Enabled           bool            `json:"enabled,omitempty"`
+	AuthorizerEnabled bool            `json:"authorizerEnabled,omitempty"`
+	SecretRef         SecretReference `json:"secretRef,omitempty"`
```

Comment:
Like mentioned in my other comment, recommend making this a pointer to a `SecretReference` and adding the `// +optional` kubebuilder annotation.

Reply:
Done